torch/distributed/_sharded_tensor/api.py

```diff
@@ -423,3 +470,71 @@ def remote_shards(self) -> Dict[int, List[rpc.RRef[Shard]]]:
                 "torch.distributed.rpc.init_rpc before creating the ShardedTensor for remote_shards support"
             )
         return self._remote_shards
+
+    def __repr__(self):
+        return str(self._metadata)
+
+    @dataclass
+    class ProcessGroupState:
+        """
+        State for ser-de of process group
+        """
+        local_rank: int
+        global_rank: int
+        local_world_size: int
+        global_world_size: int
+
+    def __getstate__(self):
+        pg_state = ShardedTensor.ProcessGroupState(
+            distributed_c10d.get_rank(self._process_group),
+            distributed_c10d.get_rank(),
+            distributed_c10d.get_world_size(self._process_group),
+            distributed_c10d.get_world_size(),
+        )
+
+        return self._local_shards, self._metadata, pg_state, self._sharding_spec
+
+    def __setstate__(self, state):
+        if not distributed_c10d.is_initialized():
+            raise RuntimeError(
+                'Need to initialize default process group using '
+                '"init_process_group" before loading ShardedTensor')
+
+        self._local_shards, self._metadata, pg_state, self._sharding_spec = state
+        self._rpc_initialized = False
```

Comment:
should we also recover this state? a sharded tensor have rpc initialized got saved, and reloading should also have rpc initialized?

Reply:
It is automatically initialized below when we call `init_rpc`. I think we might have to rework this part anyways, since it seems like users would like this behavior to be explicit where we set up RRefs only if requested via `init_rrefs` parameter in the ShardedTensor constructor.